### ISSUE TITLE
Build prerequisites are missing some components

### ISSUE THREAD

**Issue by Samonitari** [onivim/oni2#140]
Some prerequisites are not mentioned in the Readme.md

- **Python 2**
As I found out, node-gyp needs Python 2.x (don't know which is the oldest it can use). Easiest way to fix Python 3 rooted error to PREpend Python 2's path to PATH variable, when building Textmate.
- ~~**Node.js**
Maybe it sounds silly to mention it, but: even though I work in place with more than 50 programmers, I doubt there is even 5 who has it installed - we work in the medical field, so mostly C++ is the used language.~~ Yeah, it was silly, Esy's prerequisite is node.js 😳

I don't know what else is missing though, I have quite the stuff installed.

**Comment by alexmironof** [comment]
@Samonitari on Windows you need `windows-build-tools` for compiling native binaries with `node-gyp`. It's mentioned  here https://github.com/nodejs/node-gyp#on-windows

**Comment by politinsa** [comment]
As pointed out in #56 , debian users should change their /etc/apt/sources.list to get libharfbuzz-dev from sid repo.  
Apart from that, I had no issues installing oni2 on debian.

**Comment by Samonitari** [comment]
Sorry for the late answer!

First, I failed to mention in this ticket that I tried building oni2 on Windows (first)
Anyways, it can be easy to be lost in the transitive dependency chain, which is not explicit as it is.
My build steps were:

0. I had native (C++) development environment and python3 already
1. Look at the prerequisities: git was already installed, I needed esy
2. Go to esy [Get Started](https://esy.sh/docs/en/getting-started.html): realized I need npm/node,js
3. Go to Node.js: downloaded and installed the [LTS relase](https://nodejs.org/dist/v10.15.3/node-v10.15.3-x64.msi)
4. Go back to 2: installed esy with npm.
5. Do the build steps (couple times before textmate got included)
6. Do the textmate build: error msg says node-gyp fails on print command
-> quickly realize the installed python3 is called but it expects python2
7. Install python2, add it to the front of the PATH in the build cmd shell (as I don't want to break my python3 environment) -> happyness!

Nowhere in that chain I have been notified that I need Python 2 (which is frankly a bit lame, considering it'll soon be marked EOL, but that's another story, and its node-gyp's fault).
I'll admit I'm not versed in web-related programming language stacks, but I didn't even realized node-gyp has been installed somewhere (was it installed with node.js or did esy install it?), so how should I be aware of its missing dependencies?

I hope you don't get it as nitpicking (I am constantly checking out the changes of the repo, and I am thrilled by every minor progress), and maybe every tester already has node-gyp, but maybe I am not the only one to have to figure it out. A little note should not be distrubing in the prerequisites.
